lib/LDAPUserManager.php

```diff
@@ -337,16 +348,20 @@ public function deleteUser($uid) {
 	 * Change the password of a user
 	 */
 	public function setPassword($uid, $password) {
+		$userDN = $this->getUserDN($uid);
+		return $this->setPasswordForDn($userDN);
+	}
+
+	private function setPasswordForDn($userDN, $password) {
 		if(!function_exists('ldap_exop_passwd')) {
 			// since PHP 7.2 – respondToActions checked this already, this
 			// method should not be called. Double check due to public scope.
 			// This method can be removed when Nextcloud 16 compat is dropped.
 			return false;
 		}
 		try {
-			$cr = $this->ldapProvider->getLDAPConnection($uid);
-			$userDN = $this->getUserDN($uid);
-			return ldap_exop_passwd($cr, $userDN, '', $password);
+			$conn = $this->ldapConnect->getLDAPConnection();
```

Comment:
for existing users, the connection should always come from `$this->ldapProvider->getLDAPConnection($uid)`, since the function used here does not gurarantee it connects to the correct LDAP server (on multi-server setups – yes, we do not have proper support for them here yet, nevertheless it is a valid and possible caes).

Reply:
OK, so I'll leave `setPassword()` as it was before and use `ldap_exop_passwd` directly in `createUser`.